velox/expression/CastExpr.cpp

```diff
@@ -621,6 +626,59 @@ void CastExpr::applyPeeled(
   }
 }
 
+VectorPtr CastExpr::applyTimestampToVarcharCast(
+    const TypePtr& toType,
+    const SelectivityVector& rows,
+    exec::EvalCtx& context,
+    const BaseVector& input) {
+  VectorPtr result;
+  context.ensureWritable(rows, toType, result);
+  (*result).clearNulls(rows);
+  auto flatResult = result->asFlatVector<StringView>();
+  const auto simpleInput = input.as<SimpleVector<Timestamp>>();
+
+  const TimestampToStringOptions& options = hooks_->timestampToStringOptions();
+  const uint32_t rowSize = getMaxStringLength(options);
+  const auto precisionWidth = static_cast<int8_t>(options.precision);
+
+  if (StringView::isInline(rowSize)) {
```

Comment:
The estimated row size is always larger than the inline size. Do we need this branch?